File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {

        int len=0,pos=1;
        ListNode curr=head;
        

        while(curr!=null){
            curr=curr.next;
            len++;
        }

        if(len==n)
        return head.next;


        curr=head;
        while(curr!=null){
            if(len-pos==n){
                ListNode temp=curr.next.next;
                curr.next=temp;
                return head;
            }

            curr=curr.next;
            pos++;
        }
        return head;
    }
}

File: Data Structures & Algorithms/reverse-nodes-in-k-group/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* reverseKGroup(ListNode* head, int k) {
        ListNode* temp=head;
        ListNode* start=head;
        int c=0;

        while(temp!=nullptr){
            c++;
            if(c==k){
                ListNode* y = temp->next;
                temp->next=nullptr;
                ListNode* x = reverseList(start);
                c=0;
                start->next=reverseKGroup(y,k);
                return x;
            }
            temp=temp->next;
        }
        return start;

        
    }


    ListNode* reverseList(ListNode* head){
        ListNode* curr=head;
        ListNode* prev=nullptr;

        while(curr!=nullptr){
            ListNode* temp = curr->next;
            curr->next=prev;
            prev=curr;
            curr=temp;
        }
        return prev;

    }
};
